test/testAdaptors/CompoundV2AdditionalTests.t.sol

```diff
@@ -0,0 +1,1551 @@
+            0.001e18,
+            "Total assets should equal assets deposited minus swap fees."
+        );
+    }
+
+    // helper to produce the amountToBorrow or amountToWithdraw to get a certain health factor
+    // uses precision matching option A
+    function _generateAmountBasedOnHFOptionA(
+        uint256 _hfRequested,
+        address _account,
+        uint256 _borrowDecimals,
+        bool _borrow
+    ) internal view returns (uint256) {
+        // get the array of markets currently being used
+        CErc20[] memory marketsEntered;
+        marketsEntered = comptroller.getAssetsIn(address(_account));
+        PriceOracle oracle = comptroller.oracle();
+        uint256 sumCollateral;
+        uint256 sumBorrow;
+        uint256 marketsEnteredLength = marketsEntered.length;
+        // Loop to calculate total collateral & total borrow for HF calcs w/ assets we're in.
+        for (uint256 i = 0; i < marketsEnteredLength; i++) {
+            CErc20 asset = marketsEntered[i];
+            (, uint256 cTokenBalance, uint256 borrowBalance, uint256 exchangeRate) = asset.getAccountSnapshot(_account);
+            uint256 oraclePrice = oracle.getUnderlyingPrice(asset);
+            // get collateral factor from markets
+            (, uint256 collateralFactor, ) = comptroller.markets(address(asset)); // always scaled by 18 decimals
+            uint256 actualCollateralBacking = cTokenBalance.mulDivDown(exchangeRate, 1e18); // NOTE - this is the 1st key difference usage of a different scaling factor than in OptionB and CompoundV2. This means less precision but it is possibly negligible.
+            actualCollateralBacking = actualCollateralBacking.mulDivDown(oraclePrice, 1e18); // NOTE - this is the 2nd key difference usage of a different scaling factor than in OptionB and CompoundV2. This means less precision but it is possibly negligible.
+            actualCollateralBacking = actualCollateralBacking.mulDivDown(collateralFactor, 1e18); // scaling factor for collateral factor is always 1e18.
+            uint256 additionalBorrowBalance = borrowBalance.mulDivDown(oraclePrice, 1e18); // converts cToken underlying borrow to USD
+            sumCollateral = sumCollateral + actualCollateralBacking;
+            sumBorrow = additionalBorrowBalance + sumBorrow;
+        }
+
+        if (_borrow) {
+            uint256 borrowAmountNeeded = (sumCollateral.mulDivDown(1e18, _hfRequested) - sumBorrow) /
+                (10 ** (18 - _borrowDecimals)); // recall: sumBorrow = sumCollateral / healthFactor --> because specific market collateral factors are already accounted for within calcs above
+            return borrowAmountNeeded;
+
+            // uint256 borrowAmountNeeded = (sumCollateral.mulDivDown(1e18, _hfRequested) - sumBorrow);
+            // return borrowAmountNeeded;
+        } else {
+            uint256 withdrawAmountNeeded = (sumCollateral - (sumBorrow.mulDivDown(_hfRequested, 1e18)) / (10 ** (18 - _borrowDecimals)));
```

Comment:
@crispymangoes this is the helper function that is giving me issues for the two failing tests, specifically the lines of code within this else statement.